core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go

```diff
@@ -276,30 +253,46 @@ func (amc *AssetManagementContract) ClaimFungibleAsset(ctx contractapi.Transacti
             }
         }
         if err != nil {
-            log.Warn("Unable to set 'ClaimFungibleAsset' event")
-            log.Warn(err.Error())
+	    logWarnings("Unable to set 'ClaimFungibleAsset' event", err.Error())
         }
     }
     return retVal, err
 }
 
-func (amc *AssetManagementContract) UnlockAsset(ctx contractapi.TransactionContextInterface, assetAgreementSerializedProto64 string) (bool, error) {
-
-    // Decoding from base64
-    assetAgreementSerializedProto, err64 := base64.StdEncoding.DecodeString(assetAgreementSerializedProto64)
-    if err64 != nil {
-      log.Error(err64.Error())
-      return false, err64
+func (amc *AssetManagementContract) ClaimAssetUsingContractId(ctx contractapi.TransactionContextInterface, contractId, claimInfoSerializedProto64 string) (bool, error) {
+    if len(contractId) == 0 {
+        return false, logThenErrorf("empty contract id")
+    }
+    claimInfo, err := amc.validateAndExtractClaimInfo(claimInfoSerializedProto64)
+    if err != nil {
+        return false, err
     }
 
-    assetAgreement := &common.AssetExchangeAgreement{}
-    if len(assetAgreementSerializedProto) == 0 {
-        log.Error("empty asset agreement")
-        return false, fmt.Errorf("empty asset agreement")
+    // The below 'SetEvent' should be the last in a given transaction (if this function is being called by another), otherwise it will be overridden
+    retVal, err := amc.assetManagement.ClaimAssetUsingContractId(ctx.GetStub(), contractId, claimInfo)
+    if retVal && err == nil {
+        claimInfoHTLC := &common.AssetClaimHTLC{}
+        err = proto.Unmarshal(claimInfo.ClaimInfo, claimInfoHTLC)
+        if err == nil {
+            contractInfo := &common.AssetContractHTLC{
+                ContractId: contractId,
+                Claim: claimInfoHTLC,
+            }
+            contractInfoBytes, err := proto.Marshal(contractInfo)
+            if err == nil {
+                err = ctx.GetStub().SetEvent("ClaimAssetUsingContractId", contractInfoBytes)
+            }
+        }
+        if err != nil {
+	    logWarnings("Unable to set 'ClaimAssetUsingContractId' event", err.Error())
```

Comment:
We can just set this event name to `ClaimAsset`. The event receiver will see the same structure regardless of whether it contains a contract id or not, and can accordingly perform suitable actions.

Reply:
Done @VRamakrishna .